marblerun/features/secrets-management.md

```diff
@@ -2,28 +2,25 @@
 
-The Coordinator derives virtual sealing keys from respective Marble's ID and a master secret only known to the Coordinator using the [HKDF](https://tools.ietf.org/html/rfc5869) scheme. The ID is a public 128-bit value randomly generated by each Marble for itself upon startup and persisted in plaintext on local storage. Consequently, two Marbles could end up with the same ID and virtual sealing key under certain circumstances. The security implications of this need to be taken into account when using virtual sealing keys. For example, when using virtual sealing keys with AES-GCM, one must only use cryptographically random nonces.
+Next to this automatically generated certificate, you can also define additional ones  as secrets in your manifest. These are also signed by the Coordinator's intermediate CA and generated upon each launch, but do not contain a specific Marble's by default. You can include them via placeholders such as `{{ pem .Secrets.MySecretCertificate }}` in your manifest.
 
-## Shared symmetric keys
+!> Certificate secrets are always regenerated upon launch of a Marble, making them not suitable for certificate pinning.
 
-While virtual sealing keys are used by individual Marbles, MarbleRun also allows for the sharing of symmetric encryption keys between Marbles via placeholders in the manifest. The Coordinator creates these keys once and provides them as parameters to corresponding Marbles. Marbles can use shared keys for a variety of tasks, including the bulk encryption of shared data.
+## User-defined secrets
 
-As with virtual sealing keys, care has to be taken to not repeat nonces between Marbles when using shared keys with AES-GCM or similar encryption algorithms.
+In addition to automatically generated secrets, MarbleRun also allows you to define secrets which can be uploaded and updated after the initial deployment of your manifest. This can be used to store non-random sensitivate data, such as passwords, tokens, certificates, or other types of credentials that should not be stored in the manifest directly. They can be injected via the placeholders in the same way as other secret types. Changes to user-defined secrets are logged and can be audited over via the `/update` or `/secrets` HTTP REST API endpoint of the Coordinator. You can learn more about this in the section about [Managing secrets](workflows/managing-secrets).
```

Comment:
```suggestion
In addition to secrets generated by the Coordinator, MarbleRun also allows you to define secrets that can be uploaded and updated after the initial deployment of your manifest. This can be used to store sensitive data, such as passwords, tokens, certificates, or other types of credentials that should not be stored in the manifest directly. They can be injected via the placeholders in the same way as other secret types. Changes to user-defined secrets are logged and can be audited over via the `/update` or `/secrets` HTTP REST API endpoint of the Coordinator. You can learn more about this in the section about [Managing secrets](workflows/managing-secrets).
```